api/post/post_controller.go

```diff
@@ -20,23 +20,19 @@ func (c *controller) add(wr http.ResponseWriter, r *http.Request) {
 	ctx := r.Context()
 
 	body, err := ioutil.ReadAll(r.Body)
-
 	if err != nil {
 		log.Printf("Could not extract body: %s\n", err)
 		wr.WriteHeader(http.StatusBadRequest)
 		return
 	}
 
-	auth := r.Header.Get("Authorization")
-	decodedCredentials, err := base64.StdEncoding.DecodeString(auth[6:])
+	username, err := c.ce.Extract(r)
 	if err != nil {
 		log.Printf("Authorization decode error: %s", err)
```

Comment:
The credentials extractor returns Authorization decore error, and you wrap it again with the same